### PR TITLE
T/468 Fixes for Undo

### LINKED ISSUE #468: Fixes for Undo.
Umbrella issue for all bugs discovered in engine while testing undo feature.

### PR DESCRIPTION
**Pull request by scofalik**
Fixes for bugs discovered when fixing Undo feature. Fixes #468.

### CODE REVIEW

File: src/model/delta/delta.js [hunk 1]
```diff
@@ -57,6 +57,15 @@ export default class Delta {
 	}
 
 	/**
+	 * @param {Number} baseVersion
+	 */
+	set baseVersion( baseVersion ) {
+		for ( let operation of this.operations ) {
+			operation.baseVersion = baseVersion++;
```
Comment (pjasiun) [inline src/model/delta/delta.js]: It should be protected, or removed.

File: src/model/composer/deletecontents.js [hunk 2]
```diff
@@ -25,11 +25,15 @@ export default function deleteContents( batch, selection, options = {} ) {
...
-	const endPos = LivePosition.createFromPosition( selection.getFirstRange().end );
+	const selRange = selection.getFirstRange();
 
-	// 1. Remove the contents.
-	batch.remove( selection.getFirstRange() );
+	const startPos = selRange.start;
+	const endPos = LivePosition.createFromPosition( selRange.end );
+
+	// 1. Remove the contents if there are any.
+	if ( !selRange.isEmpty ) {
```
Comment (pjasiun) [inline src/model/composer/deletecontents.js]: Missing else path coverage.

File: src/model/delta/delta.js (hunk 1 above)
Comment (scofalik) [inline src/model/delta/delta.js]: Okay, I'll change it but in next iteration of changes for Undo this code will be removed anyway.

**Comment by scofalik** [comment]
Fixed

**Pull request merged**